### PR TITLE
A widget to display a table for EELS edges

### PR DESCRIPTION
**Pull request by ptim0626**
This include a widget to display a table for EELS edges, from an energy within selected by a span selector. Edges can be marked by selecting them in the list. See hyperspy/hyperspy#2418.

### CODE REVIEW

File: hyperspy_gui_ipywidgets/tools.py [hunk 1]
```diff
@@ -169,22 +167,9 @@ def print_edges_table_ipy(obj, **kwargs):
...
-    table = ipywidgets.interactive_output(obj.show_edges_table, 
-                                          {'x0': left,
-                                           'x1': right,
-                                           'only_major': major,
-                                           'update': update,
-                                           'order': order,
-                                           'active_edges': edges_list,
-                                           'complementary': complmt}
-                                          )
+    update = ipywidgets.Button(description='Update', layout={'width': 'initial'})
```
[the remark this answers is not in the document]
Comment (ptim0626) [reply hyperspy_gui_ipywidgets/tools.py]: `Refresh table` looks good.

File: hyperspy_gui_ipywidgets/tools.py [hunk 2]
```diff
@@ -147,6 +147,140 @@ def on_close_clicked(b):
...
+                                )
+    update = ipywidgets.Button(description='Update', layout={'width': 'initial'})
+    gb = ipywidgets.GridBox(layout=ipywidgets.Layout(
+            grid_template_columns="70px 125px 75px 250px")) 
+    help = ipywidgets.HTML(
+        "Click on the signal figure and drag to the right to select a signal "
+        "range. Drag the rectangle or change its border to display edges in "
+        "different signal range. Select edges to show their positions " 
+        "on the signal.",)
+    help = ipywidgets.Accordion(children=[help])
```
[the remark this answers is not in the document]
Comment (ptim0626) [reply hyperspy_gui_ipywidgets/tools.py]: Suggestion taken

File: hyperspy_gui_ipywidgets/tools.py [hunk 3]
```diff
@@ -147,6 +147,140 @@ def on_close_clicked(b):
...
     }
 
+@add_display_arg
+def print_edges_table_ipy(obj, **kwargs):
+    # Define widgets
+    wdict = {}
+    axis = obj.axis
+    style_d = {'description_width': 'initial'}
+    layout_d = {'width': '50%'}
+    left = ipywidgets.FloatText(disabled=True, layout={'width': '25%'})
```
[the remark this answers is not in the document]
Comment (ptim0626) [reply hyperspy_gui_ipywidgets/tools.py:157]: It seems the `FloatText` object doesn't respect the `readout_format` as it doesn't contain it as attribute and setting `readout_format='.5g'` or whatever have no effect on the displayed text. Bug in ipywidgets?

[the remark this answers is not in the document]
Comment (ericpre) [reply hyperspy_gui_ipywidgets/tools.py:157]: Indeed, it seems that this isn't implemented for the `FloatText`, see for example https://github.com/jupyter-widgets/ipywidgets/issues/650 for discussion on the topic.

**Comment by ericpre** [comment]
@ptim0626, can you please merge with upstream to solve the conflict?

**Comment by ericpre** [comment]
Now that the tests are passing it would be good to test the EELS table widget!

**Comment by francisco-dlp** [comment]
A test is failing.

**Comment by ptim0626** [comment]
@ericpre @francisco-dlp  It failed at a `AttributeError` when `random_state = <mtrand.RandomState object at 0x7f935fbb0ee8>` trying to access `random` attribute when creating an artificial EELS dataset. This doesn't happen in the `hyperspy` repo or my repo (tests passed there), as the `random_state` is a `RandomState(MT19937) ` object, any idea what's going on?

**Comment by francisco-dlp** [comment]
@tjof2 should be able to help with this.

**Comment by tjof2** [comment]
@francisco-dlp I'm away from a PC most of the day. The code that it errors needs to be `random_state.uniform` rather than `random_state.random`. can you fix it? Cc. @ericpre

**Comment by tjof2** [comment]
i.e. here https://github.com/hyperspy/hyperspy/blob/f90e6406adcc98e4a4ed6375a0fa26e383245dc1/hyperspy/datasets/artificial_data.py#L176

**Comment by francisco-dlp** [comment]
Thanks @tjof2, I've tried that but it [didn't fix it](https://travis-ci.org/github/hyperspy/hyperspy_gui_ipywidgets/jobs/714997112#L685). It doesn't help that I can't reproduce it in my computer either...

**Comment by francisco-dlp** [comment]
I've applied the same fix at 3 other places, let's see if now it works.

**Comment by francisco-dlp** [comment]
That fixed it, thanks @tjof2. I'll leave @ericpre decide if this is ready.

**Comment by ericpre** [comment]
Yes, it looks good to me!

**Pull request merged**